### PR TITLE
Run string cast on $options

### PR DESCRIPTION
**Pull request by Fonger**
**Summary**

Apply the fix in #1462 $conditionalHandlers for String schema #7800 

https://github.com/Automattic/mongoose/blob/c08002049442d0f87c99819529177f9ba148048b/lib/schema/array.js#L439-L442

Maybe we can throw an error if `$options` is not of type string.

**Examples**

See test

Close&Reopen to re-trigger travis-ci test.

### CODE REVIEW

File: lib/schema/string.js [hunk 1]
```diff
@@ -577,7 +577,7 @@ SchemaString.prototype.$conditionalHandlers =
       $gte: handleSingle,
       $lt: handleSingle,
       $lte: handleSingle,
-      $options: handleSingle,
```
Comment (vkarpov15) [inline lib/schema/string.js]: Why is this change necessary? `handleSingle()` will cast it to a string, no?

Comment (Fonger) [reply lib/schema/string.js]: handleSingle() will cast it to a string, but also apply uppercase/lowercase setter to reproduce #7800.
Change to String here like the way `SchemaArray` handles `$options`
